modules/tensor_mechanics/src/materials/ComputeFiniteStrainElasticStress.C

```diff
@@ -55,12 +50,9 @@ ComputeFiniteStrainElasticStress::computeQpStress()
   // Calculate the stress in the intermediate configuration
   RankTwoTensor intermediate_stress;
 
-  // Check if the elasticity tensor has changed values
-  if (!_is_elasticity_tensor_guaranteed_constant_in_time)
-    intermediate_stress =
-        _elasticity_tensor[_qp] * (_elastic_strain_old[_qp] + _strain_increment[_qp]);
-  else
-    intermediate_stress = _stress_old[_qp] + _elasticity_tensor[_qp] * _strain_increment[_qp];
+  intermediate_stress =
+      _elasticity_tensor[_qp] * (_elastic_strain_old[_qp] + _strain_increment[_qp]);
+  addQpInitialStress(intermediate_stress);
```

Comment:
Is there a potential problem with keeping the initial stress in the same current configuration with the stress?  The initial stress is always given in the reference configuration, but we only apply the incremental rotation to this initial stress quantity.

Reply:
OK, i have completely re-written this stuff because of the finite-strain rotations.  I've tried to have no performance or memory penalties in the case where initial_stress is not given, because i think that's the most common use-case in MOOSE